connection_test.go

```diff
@@ -1260,6 +1260,64 @@ func TestLastActivityTimePings(t *testing.T) {
 	})
 }
 
+func TestSendBufferSize(t *testing.T) {
+	opts := testutils.NewOpts().SetSendBufferSize(512).SetSendBufferSizeOverrides([]SendBufferSizeOverride{
+		{"abc", 1024},
+		{"abcd", 2048},
+		{"xyz", 3072},
+	})
+	tests := []struct {
+		processName          string
+		expectSendChCapacity int
+	}{
+		{
+			processName:          "abc",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "abcd",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "bcd",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "dabc",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "dabcd",
+			expectSendChCapacity: 512,
+		},
+		{
+			processName:          "abcde",
+			expectSendChCapacity: 1024,
+		},
+		{
+			processName:          "xyzabc",
+			expectSendChCapacity: 3072,
+		},
+	}
+	for _, tt := range tests {
+		testutils.WithTestServer(t, opts.NoRelay(), func(tb testing.TB, ts *testutils.TestServer) {
+			client := ts.NewClient(opts.SetProcessName(tt.processName))
+
+			// Send an 'echo' to establish the connection.
+			testutils.RegisterEcho(ts.Server(), nil)
+			require.NoError(t, testutils.CallEcho(client, ts.HostPort(), ts.ServiceName(), nil))
+
+			// WithTestSever will test with and without relay.
+			if opts.DisableRelay {
+				assert.Equal(t, tt.expectSendChCapacity, getConnection(t, ts.Server(), inbound).SendChCapacity)
+			} else {
+				assert.Equal(t, tt.expectSendChCapacity, getConnection(t, ts.Relay(), inbound).SendChCapacity)
```

Comment:
this `else` case should not be needed if we're using `NoRelay`

Reply:
Actually this was a mistake. We do want to test with and without relay. Changed the if condition to to `ts.HasRelay()`.